schema.ts

```diff
@@ -1280,6 +1292,18 @@ export enum PrincipalType {
 export interface UserPrincipal {
   type: PrincipalType.User;
   userId: string | number;
+  userIdType?: UserIdType;
+}
+
+/**
+ * The type of group ID that is being used in a {@link GroupPrincipal}.
+ *
+ * TODO(sam): Unhide this
+ * @hidden
+ */
+export enum GroupIdType {
+  GoogleGroup = 'googleGroup',
```

Comment:
A Google Group doesn't feel like a type of ID, but rather a type of group itself. A Google Group in fact has multiple IDs, both a numeric ID and an associated email address.